docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+Suffice to say that each `ConnectionEnd` has associated the `State` of both the initiator and responder process.
+We will declare this data types later (see [2.2 Protocol](#22-protocol)).
+
+We now restate the connection handshake problem in a slightly more precise (or lower level) formulation:
+A connection handshake enables two processes to agree on a consistent view of their state, to verify each other's state, and to allocate a new connection.
+
+
+###### Guarantees
+We refine the safety guarantee **[CH/Seq/1/Agreement]**  (defined in the [sequential problem statement](#sequential-problem-statement)) by splitting this into four complementary safety properties:
+
+- **[CH/Proto/1/Uniqueness]**
+If a process finishes a handshake for a `ConnectionEnd` `c`, no handshake has previously finished for `c`.
+
+- **[CH/Proto/2/Validity]**
+If either of the responder or initiator agree to a `ConnectionEnd` `c`, then the initiator proposed `c`.
+
+- **[CH/Proto/3/StateConsistency]**
+If process `i` establishes a connection with some process `r`, then process `i` has the `State` `si` representing process `r` and this state is consistent with the actual state of `r`.
+
+- **[CH/Proto/4/StateVerification]**
+If the processes establish a connection  for their respective `State` `si` (state of initiator) and `sr` (state of responder), then the initiator verified `sr` and the responder verified `si`.
+
+The liveness property below is functionally identical to the **[CH/Seq/2/Termination]** property we defined earlier:
+
+- **[CH/Proto/5/Termination]** If the initiator and responder are correct, then the handshake eventually terminates (i.e., both processes allocate a new `ConnectionEnd` object).
+
+**Remarks**.
+
+- Uniqueness.
+To guarantee uniqueness, we can assume there is a predicate `unique(v)` that processes invoke to check the uniqueness of input value `v`; this predicate returns true if the input value `v` is unique and otherwise returns false.
+
+- State verification.
+Like uniqueness, processes verify each other's state by calling on a predicate `verify(s)` that returns true iff the input state `s` passes verification.
+
+- State consistency.
+This property is for ensuring that processes are not out of sync.
+This property complements state verification: while verification ensures authenticity on the state of another process (_who_ is the other process), consistency ensures freshness on the state of another process (_what_ is the other process).
+We assume there is a predicate `consistent(s)` that returns true iff the input state `s` is consistent.
+
+### 2.2 Protocol
+
+> The following section is work-in-progres..
+
+The CH protocol consists of four datagrams that the initiator and responder exchange.
+We enumerate the names of these datagrams in the following code snippet.
+
+```golang
+type ConnHandshakeInit struct {  }
+type ConnHandshakeTry struct {  }
+type ConnHandshakeAck struct {  }
+type ConnHandshakeConfirm struct {  }
+```
```

Comment:
Should we use names closer to the IBC spec? Maybe they are not perfect but I think in this case using `ConnOpenInit`, etc. it's not too confusing. 
Also, the connection spec also covers the closing handshake.

Reply:
If we don't fix names now, we'll probably never do it and remain stuck with them. Note that these are protocol-level modifications, so should be backwards-compatible with the Go implementation. But I have no strong preference, it is just a proposal to make things more consistent.